redaxo/src/core/lib/login/user.php

```diff
@@ -50,7 +50,28 @@ public static function get(int $id): ?self
         return static::getInstance($id, static function (int $id) {
             $sql = rex_sql::factory()->setQuery('SELECT * FROM '.rex::getTable('user').' WHERE id = ?', [$id]);
 
-            return $sql->getRows() ? new static($sql) : null;
+            if ($sql->getRows()) {
+                $user = new static($sql);
+                static::addInstance('login_' . $user->getLogin(), $user);
+                return $user;
+            }
+
+            return null;
+        });
+    }
+
+    public static function getByLogin(string $login): ?self
+    {
+        return static::getInstance('login_' . $login, static function (string $login) {
+            $sql = rex_sql::factory()->setQuery('SELECT * FROM '.rex::getTable('user').' WHERE login = ?', [$login]);
+
+            if ($sql->getRows()) {
+                $user = new static($sql);
+                static::addInstance($user->getId(), $user);
```

Comment:
und hier müsst es vermutlich anders sein:
```suggestion
                static::addInstance('login_' . $user->getLogin(), $user);
```